github/githubapp.go

```diff
@@ -0,0 +1,99 @@
+package github
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"path"
+	"strconv"
+	"time"
+
+	"github.com/golang-jwt/jwt/v5"
+)
+
+type GitHubApp struct {
+	ClientId       string `json:"client_id"`
+	InstallationId int    `json:"installation_id"`
+	PrivateKey     string `json:"private_key"` // SENSITIVE
+}
+
+func (app *GitHubApp) IsZero() bool {
+	return *app == GitHubApp{}
+}
+
+// generateJWTtoken returns a signed JWT token used to authenticate as GitHub App
+func generateJWTtoken(clientId, privateKey string) (string, error) {
+	key, err := jwt.ParseRSAPrivateKeyFromPEM([]byte(privateKey))
+	if err != nil {
+		return "", fmt.Errorf("could not parse private key: %w", err)
+	}
```

Comment:
Why are we parsing it that late instead of while validating the config?

Reply:
@odormond do you mind if I keep it as it is for this PR, and try to propose another separate PR addressing this in a more focused way?

Reply:
Fine for me.